src/main/java/se/liu/ida/hefquin/engine/query/impl/QueryPatternUtils.java

```diff
@@ -170,18 +171,18 @@ public static int getNumberOfBNodeOccurrences( final BGP bgp ) {
 	 * Returns the number of occurrences of variables in the given graph
 	 * pattern. If the same variable occurs multiple times, each occurrence
 	 * is counted.
-	 * 
+	 *
 	 * If the given pattern is a {@link TriplePattern}, this function returns
 	 * the result of {@link #getNumberOfVarOccurrences(TriplePattern)}.
 	 * Similarly, if the given pattern is a {@link BGP}, this function
 	 * returns the result of {@link #getNumberOfVarOccurrences(BGP)}.
 	 */
 	public static int getNumberOfVarOccurrences( final SPARQLGraphPattern queryPattern ) {
-		if ( queryPattern instanceof TriplePattern ) {
-			return getNumberOfVarOccurrences( (TriplePattern) queryPattern );
+		if ( queryPattern.asJenaOp() instanceof Triple) {
```

Comment:
This will never be true. `queryPattern.asJenaOp()` returns a Jena `Op` object but Jena's `Triple` class does not implement `Op`. Probably you meant `OpTriple` instead?

Something else: I would also like to keep the original if-branch (i.e., `queryPattern instanceof TriplePattern`) because that fires immediately for `TriplePattern` objects. In fact, if you look at `TriplePatternImpl`, you will notice that `asJenaOp()` is not even implemented there.